test/functional/apps/discover/_field_data.ts

```diff
@@ -36,6 +36,7 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
       await esArchiver.load('discover');
       await kibanaServer.uiSettings.replace({
         defaultIndex: 'logstash-*',
+        'discover:searchFieldsFromSource': true,
```

Comment:
I think it's worth leaving this test as-is. The exact copy with retrieving fields from the fields API is `_field_date_with_fields_api`.